src/d3a/models/area/event_dispatcher.py

```diff
@@ -123,14 +129,14 @@ def event_listener(self, event_type: Union[MarketEvent, AreaEvent], **kwargs):
             self.area.strategy.event_on_disabled_area()
 
     @staticmethod
-    def create_agent_object(owner, higher_market, lower_market, is_spot_market):
+    def create_agent_object(owner, higher_market, lower_market, market_type):
```

Comment:
First of all, double quotes does not remove the need to import the class of the type hint. Even in this case we need to do `from d3a.models.markets import Market` in order for python interpreter to not complain. As a followup to that, if I import the Area class in this module I will end up with a circular import error. I can avoid it by using this https://stackoverflow.com/a/39757388, but conditional imports only for type hints is very ugly IMO and IIRC we have discussed to not do it and simply not add type hints in these cases. TLDR, I will not add type hint for the owner argument

Reply:
I haven't tested this, but [PEP 484 (Forward references)](https://www.python.org/dev/peps/pep-0484/#forward-references) describes a way to overcome circular imports:
> A common use for forward references is when e.g. Django models are needed in the signatures. Typically, each model is in a separate file, and has methods taking arguments whose type involves other models. Because of the way circular imports work in Python, it is often not possible to import all the needed models directly [...]
The solution is to switch to module-only imports and reference the models by their `_module_._class_ name`.

So for example:
```python
from models import b
class A(Model):
    def foo(self, b: 'b.B'):
        pass
```

Reply:
I also do not like this solution actually, mainly due to the fact that module-based imports can be very broad, and also because they can lead to long/verbose types (e.g. one type hint could be d3a.models.area.Area which is overly verbose). More specifically in this example, this would not solve our problem because the Area class is on the d3a.models.area.__init__ module. Therefore when one does `from d3a.models import area` , the Area class will also be imported thus causing the circular import, despite the fact that area.Area would be used. 
The correct way to solve this is not by trying to cheat the Python interpreter, but by really refactoring the code in order to avoid such occurrences in the first place (by moving classes around files in a way that circular imports are avoided). However this is something that should be added as a separate task and not as part of this ticket.